core/cache/controller.go

```diff
@@ -364,7 +385,14 @@ func (c *Controller) ensureModel(modelUUID string) *Model {
 
 	model, found := c.models[modelUUID]
 	if !found {
-		model = newModel(c.metrics, newPubSubHub(), c.manager.new())
+		initializer := &initializeState{&c.initializing}
```

Comment:
Based on what Harry says, its this also a case of unprotected access to "initializing"

Reply:
The address of initializing never changes, and this is within a lock anyway.